accord-core/src/main/java/accord/local/RedundantBefore.java

```diff
@@ -924,6 +936,22 @@ public boolean isShardOnlyApplied(TxnId txnId, Unseekables<?> participants)
         return foldl(participants, Bounds::isShardOnlyApplied, false, txnId);
     }
 
+    public boolean isUnsafeBefore(TxnId txnId, Unseekables<?> participants)
+    {
+        // Definitely safe
+        if (maxUnsafeBefore == TxnId.NONE)
+            return false;
+
+        // TODO: Problem is that, logically, we do know that transactions after maxUnsafeBefore (txnId.compareTo(maxUnsafeBefore) > 0)
```

Comment:
Is this still true? Because if so we need to understand it, and using a weaker comparison here shouldn't have any bearing on that fact, as it should only decide whether we invoke the below foldl which provides the more precise answer?

Reply:
It was, yes, since I was hunting down a couple more burn test bugs

```
        // Maybe unsafe
        if (participants == null)
            return txnId.compareTo(maxLocallyIncomplete) > 0;
```

should be correct now.